discovery-provider/src/tasks/index_solana_plays.py

```diff
@@ -0,0 +1,305 @@
+SECP_PROGRAM = "KeccakSecp256k11111111111111111111111111111"
+SLEEP_TIME = 1
+
+# Maximum number of batches to process at once
+TX_SIGNATURES_MAX_BATCHES = 20
+
+# Last N entries present in tx_signatures array during processing
+TX_SIGNATURES_RESIZE_LENGTH = 10
+
+logger = logging.getLogger(__name__)
+
+# Parse signed data message from each transaction submitted to
+# Audius TrackListenCount program
+def parse_instruction_data(data):
+    decoded = base58.b58decode(data)[1:]
+
+    user_id_length = int.from_bytes(decoded[0:4], "little")
+    user_id_start, user_id_end = 4, 4 + user_id_length
+    user_id = int(decoded[user_id_start:user_id_end])
+
+    track_id_length = int.from_bytes(decoded[user_id_end:user_id_end + 4],
+                                     "little")
+    track_id_start, track_id_end = user_id_end + 4, user_id_end + 4 + track_id_length
+    track_id = int(decoded[track_id_start:track_id_end])
+
+    source_length = int.from_bytes(decoded[track_id_end:track_id_end + 4],
+                                   "little")
+    source_start, source_end = track_id_end + 4, track_id_end + 4 + source_length
+    source = str(decoded[source_start:source_end], 'utf-8')
+
+    timestamp = int.from_bytes(decoded[source_end:source_end + 8], "little")
+
+    return user_id, track_id, source, timestamp
+
+# Retry 5x until a tx 'result' is found with valid contents
+# TODO: Clarify this unavailabilty
+def get_sol_tx_info(solana_client, tx_sig):
+    retries = 5
+    while retries > 0:
+        try:
+            tx_info = solana_client.get_confirmed_transaction(tx_sig)
+            if tx_info["result"] is not None:
+                return tx_info
+        except Exception as e:
+            logger.error(f"index_solana_plays.py | Error fetching tx {tx_sig}, {e}", exc_info=True)
+        retries -= 1
+        logger.error(f"index_solana_plays.py | Retrying tx fetch: {tx_sig}")
+
+def parse_sol_play_transaction(session, solana_client, tx_sig):
+    try:
+        logger.info(f"index_solana_plays.py | Parsing {tx_sig}")
+        tx_info = get_sol_tx_info(solana_client, tx_sig)
+        logger.info(
+            f"index_solana_plays.py | Got transaction: {tx_sig} | {tx_info}"
+        )
+        if SECP_PROGRAM in tx_info["result"]["transaction"]["message"][
+                "accountKeys"]:
+            audius_program_index = tx_info["result"]["transaction"]["message"][
+                "accountKeys"].index(TRACK_LISTEN_PROGRAM)
+            for instruction in tx_info["result"]["transaction"]["message"][
+                    "instructions"]:
+                if instruction["programIdIndex"] == audius_program_index:
+                    tx_slot = tx_info['result']['slot']
+                    user_id, track_id, source, timestamp = parse_instruction_data(
+                        instruction["data"])
+                    created_at = datetime.datetime.utcfromtimestamp(timestamp)
+
+                    logger.info("index_solana_plays.py | "
```

Comment:
agreed, maybe this is overkill? I would like to ship v1 with this and if we see that they are unnecessary we can decrease? if we feel strongly though i am open to decreasing them

Reply:
can we consolidate some of these logs if possible? i think i counted like 6 logs per play, so if we have an avg of 250k plays a day that's 1.5M a day. our total discovery logs are like 2.5 million, so that's like increasing our existing daily logs by 60% per node. that means our loggly subscription needs to be increased by like 2/3 to support this